src/lib.rs

```diff
@@ -1251,11 +1251,7 @@ mod node {
         }
 
         pub async fn with_options(opts: IpfsOptions) -> Self {
-            let span = Some(Span::current());
-
-            let (ipfs, fut): (Ipfs<TestTypes>, _) = UninitializedIpfs::new(opts, span)
-                .in_current_span()
-                .await
+            let (ipfs, fut): (Ipfs<TestTypes>, _) = UninitializedIpfs::new(opts)
```

Comment:
the only problem I see left here is that the method would need to be generic over `<Types: IpfsTypes>` instead of assigning the `TestTypes`.